### PR TITLE
Add sourcecode to the community build

### PR DESCRIPTION
**Pull request by alexarchambault**
Corresponding branch: https://github.com/lihaoyi/sourcecode/tree/dotty

### CODE REVIEW

File: src/test/scala/dotty/communitybuild/CommunityBuildTest.scala [hunk 1]
```diff
@@ -74,4 +74,9 @@ class CommunityBuildTest extends BuildTest {
     gitUrl = "https://github.com/dotty-staging/pdbp",
     command = "compile"
   )
+
+  @Test def sourcecode = project(
+    gitUrl = "https://github.com/lihaoyi/sourcecode",
+    command = "++0.10.0 test"
```
Comment (allanrenucci) [inline src/test/scala/dotty/communitybuild/CommunityBuildTest.scala]: The community build will specify the scala version itself

Comment (alexarchambault) [reply src/test/scala/dotty/communitybuild/CommunityBuildTest.scala]: Fixed.

File: src/test/scala/dotty/communitybuild/CommunityBuildTest.scala [hunk 2]
```diff
@@ -74,4 +74,9 @@ class CommunityBuildTest extends BuildTest {
     gitUrl = "https://github.com/dotty-staging/pdbp",
     command = "compile"
   )
+
+  @Test def sourcecode = project(
+    gitUrl = "https://github.com/lihaoyi/sourcecode",
+    command = "test"
```
[the remark this answers is not in the document]
Comment (alexarchambault) [reply src/test/scala/dotty/communitybuild/CommunityBuildTest.scala]: Because of the `!` at the end of the `++…` command…

Should be fixed.

**Comment by allanrenucci** [comment]
@alexarchambault Can you investigate the failures? Is it legitimate or a bug in Dotty?

Also note that you might be affected by changes to the API: lampepfl/dotty#5474

**Comment by alexarchambault** [comment]
I think I see what it is (it doesn't take into account sources under `src/main/scala-0.10`, as it's `0.12.…` here).

**Comment by allanrenucci** [comment]
I suppose you can do something similar to what we do in the Dotty build: https://github.com/lampepfl/dotty/blob/1ba082a9639e8d8b7d28c4a6867cd8b7c5c4f0c3/project/Build.scala#L777-L785

**Comment by alexarchambault** [comment]
Ok, I just did https://github.com/lihaoyi/sourcecode/pull/62.

**Comment by alexarchambault** [comment]
Now the error is legit (API changes between `0.10` and `0.12`). Looking into it…

**Comment by alexarchambault** [comment]
https://github.com/lihaoyi/sourcecode/pull/63 should fix that.

**Comment by allanrenucci** [comment]
Awesome! Thanks @alexarchambault

**Pull request merged**